### ISSUE TITLE
Boss.Find fails to find element in iframes

### ISSUE THREAD

**Issue by atifaziz** [StephenCleary/BrowserBoss#5]
[`Boss.Find`](https://github.com/StephenCleary/BrowserBoss/blob/77421f7f1b2d13c977d381a781be66218f2d0f8d/Nito.BrowserBoss/Nito.BrowserBoss/Boss.cs#L89-L96) (and other similar methods) that eventually relies on the algorithm in [`TryFind` for a collection of finders](https://github.com/StephenCleary/BrowserBoss/blob/77421f7f1b2d13c977d381a781be66218f2d0f8d/Nito.BrowserBoss/Nito.BrowserBoss/Finders/FindExtensions.cs#L47-L73) fails when looking for an element in an `iframe`. While the algorithm is designed to search recursively through `iframe` elements, it fails because [the driver is always instructed to switch to the default content on line 57](https://github.com/StephenCleary/BrowserBoss/blob/77421f7f1b2d13c977d381a781be66218f2d0f8d/Nito.BrowserBoss/Nito.BrowserBoss/Finders/FindExtensions.cs#L57).

**Comment by atifaziz** [comment]
LINQ query to reproduce:

``` c#
Boss.StartChrome(hideCommandWindow: false);
Boss.Url = "http://jsfiddle.net/n5upnvoa/1/";
Boss.Click(".//a[@href='http://www.iana.org/domains/example']");
```

The [JSFiddle page used by the query](http://jsfiddle.net/n5upnvoa/1/) contains a single `iframe` element:

``` html
<iframe src="https://www.example.com/" width="600" height="240"></iframe>
```

The content of the www.example.com reads:

> # Example Domain
> 
> This domain is established to be used for illustrative examples in documents. You may use this domain in examples without prior coordination or asking for permission.
> 
> [More information...](http://www.iana.org/domains/example)

First try against [version 2.0.1 on NuGet](https://www.nuget.org/packages/Nito.BrowserBoss/2.0.1) and it should fail with an `InvalidDataException` and a message reading:

```
Could not find elements matching .//a[@href='http://www.iana.org/domains/example']
```

The stack trace of exception reads:

```
   at Nito.BrowserBoss.Session.FindElements(ISearchContext context, String searchText)
   at Nito.BrowserBoss.Session.FindElements(String searchText)
   at Nito.BrowserBoss.Session.Click(String searchText)
   at Nito.BrowserBoss.Boss.Click(String searchText)
   at UserQuery
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()`
```

Then try against a build of PR #6 and it should find and click the link within the iframe.

**Comment by CrystaletResearch** [comment]
I’m seeing the same issue and the fix from @atifaziz seems to do the trick. Any chance of this being merged into the code base and a new release issued?

**Comment by StephenCleary** [comment]
Fixed in v2.0.2.
